pkg/pf/tests/schema_and_program_test.go

```diff
@@ -413,3 +419,135 @@ outputs:
+						return map[int64]resource.StateUpgrader{
+							0: {
+								StateUpgrader: func(ctx context.Context, usr1 resource.UpgradeStateRequest, usr2 *resource.UpgradeStateResponse) {
+									var usr1V0 lifecycleRuleFilterModelV0
+									usr2.Diagnostics.Append(usr1.State.Get(ctx, &usr1V0)...)
+									usr2.Diagnostics.Append(usr2.State.Set(ctx, lifecycleRuleFilterModel{
+										ObjectSizeGreaterThan: stringToInt64Legacy(ctx, usr1V0.ObjectSizeGreaterThan, &usr2.Diagnostics),
+									})...)
+								},
+								PriorSchema: &rschema.Schema{
+									Version: 0,
+									Attributes: map[string]rschema.Attribute{
+										"object_size_greater_than": rschema.StringAttribute{
+											Optional: true,
+										},
+									},
+								},
+							},
+						}
+					},
+					ResourceSchema: rschema.Schema{
+						Attributes: map[string]rschema.Attribute{
+							"object_size_greater_than": rschema.Int64Attribute{
+								Optional: true,
+								Computed: true, // Because of Legacy value handling
+								PlanModifiers: []planmodifier.Int64{
+									int64planmodifier.UseStateForUnknown(),
+								},
+							},
+						},
+					},
+				}),
+			},
+		})
+
+	prov := provBuilder.ToProviderInfo()
+
+	program := `
+name: test
+runtime: yaml
+resources:
+    mainRes:
+        type: testprovider:index:Test`
+
+	pt, err := pulcheck.PulCheck(t, prov, program)
+	require.NoError(t, err)
+	pt.ImportStack(t, apitype.UntypedDeployment{
+		Version: 3,
+		Deployment: []byte(`{
+		"manifest": {
+				"time": "2025-02-20T14:09:00.155613-05:00",
+				"magic": "0cfd49ecb2b79ab5c815533dd5e24026f84295ad05c68df3861ea07ea846919a",
+				"version": "v3.150.0"
+		},
+		"resources": [
+				{
+						"urn": "urn:pulumi:test::test::pulumi:pulumi:Stack::test-test",
+						"custom": false,
+						"type": "pulumi:pulumi:Stack",
+						"created": "2025-02-20T19:09:00.146543Z",
+						"modified": "2025-02-20T19:09:00.146543Z"
+				},
+				{
+						"urn": "urn:pulumi:test::test::pulumi:providers:testprovider::default",
+						"custom": true,
+						"id": "127dc091-46cd-41f1-a3a9-ccdeca036b02",
+						"type": "pulumi:providers:testprovider",
+						"created": "2025-02-20T19:09:00.151776Z",
+						"modified": "2025-02-20T19:09:00.151776Z"
+				},
+				{
+						"urn": "urn:pulumi:test::test::testprovider:index/test:Test::mainRes",
+						"custom": true,
+						"id": "test-id",
+						"type": "testprovider:index/test:Test",
+						"inputs": {
+						},
+						"outputs": {
+								"id": "test-id",
+								"objectSizeGreaterThan": ""
+						},
+						"parent": "urn:pulumi:test::test::pulumi:pulumi:Stack::test-test",
+						"provider": "urn:pulumi:test::test::pulumi:providers:testprovider::default::127dc091-46cd-41f1-a3a9-ccdeca036b02",
+						"propertyDependencies": {
+								"objectSizeGreaterThan": []
+						},
+						"created": "2025-02-20T19:09:00.154504Z",
+						"modified": "2025-02-20T19:09:00.154504Z"
+				}
+		],
+		"metadata": {}
+}`),
+	})
+
+	pt.Up(t)
```

Comment:
This will fail with

```
        Diagnostics:
          testprovider:index:Test (mainRes):
            error: objectEncoder failed on property "object_size_greater_than": Expected a Number

```